File: TypeScript/src/session2.td/agregation/NatDeleguantEtat.js
"use strict";
exports.__esModule = true;
var NatDeleguantEtat = /** @class */ (function () {
    function NatDeleguantEtat(etat) {
        this._etat = etat;
    }
    NatDeleguantEtat.prototype.etat = function () {
        return this._etat;
    };
    NatDeleguantEtat.prototype.val = function () {
        return this._etat.val();
    };
    NatDeleguantEtat.prototype.estNul = function () {
        return this._etat.val() == 0;
    };
    NatDeleguantEtat.prototype.predecesseur = function () {
        return this.creerNatAvecEtat(this.etat().predecesseur());
    };
    NatDeleguantEtat.prototype.chiffre = function (n) {
        return this._etat.chiffre(n);
    };
    NatDeleguantEtat.prototype.taille = function () {
        return this._etat.taille();
    };
    NatDeleguantEtat.prototype.creerNatAvecValeur = function (n) {
        return this.creerNatAvecEtat(this._etat.creerNatAvecValeur(n));
    };
    NatDeleguantEtat.prototype.creerZero = function () {
        return this.creerNatAvecEtat(this._etat.creerZero());
    };
    NatDeleguantEtat.prototype.creerSuccesseur = function (predecesseur) {
        return this.creerNatAvecValeur(predecesseur.val() + 1);
    };
    NatDeleguantEtat.prototype.creerNatAvecRepresentation = function (s) {
        return this.creerNatAvecEtat(this._etat.creerNatAvecRepresentation(s));
    };
    return NatDeleguantEtat;
}());
exports.NatDeleguantEtat = NatDeleguantEtat;

File: Java/src/session2/td/agregation/NatDeleguantEtat.java
package session2.td.agregation;

public abstract class NatDeleguantEtat  implements Nat{

    protected EtatNaturelPur etat;

    public NatDeleguantEtat(EtatNaturelPur etat) {
        this.etat = etat;
    }

    public EtatNaturelPur etat(){
        return this.etat;
    }
    public int val(){
        return this.etat.val();
    }

    public boolean estNul(){
        return this.etat.val() == 0;
    }

    public Nat predecesseur(){
        return this.creerNatAvecEtat(this.etat().predecesseur());
    }

    @Override
    public int chiffre(int n) {
        return this.etat.chiffre(n);
    }

    @Override
    public int taille() {
        return this.etat.taille();
    }
    @Override
    public Nat creerNatAvecValeur(int n) {
        return this.creerNatAvecEtat(this.etat.creerNatAvecValeur(n));
    }
    @Override
    public Nat creerZero() {
        return this.creerNatAvecEtat(this.etat.creerZero());
    }
    @Override
    public Nat creerSuccesseur(Nat predecesseur) {
        return this.creerNatAvecValeur(predecesseur.val() + 1);
    }
    @Override
    public Nat creerNatAvecRepresentation(String s) {
        return this.creerNatAvecEtat(this.etat.creerNatAvecRepresentation(s));
    }



}
